src/mame/drivers/minitel_2_rpic.cpp

```diff
@@ -200,25 +230,66 @@ void minitel_state::port1_w(uint8_t data)
 void minitel_state::port3_w(uint8_t data)
 {
 	LOG("port_w: write %02X to PORT3\n", data);
+
+	m_serport->write_txd(BIT(data, PORT_3_SER_TXD_BIT));
+
 	port3 = data;
 }
 
+
+
+void minitel_state::update_modem_state()
+{
+	// 1300 Hz tone detection :  PORT_1_MDM_RTS = 1, CTRL_REG_DTMF = 0, CTRL_REG_MCBC = 0
+	if(  BIT( last_ctrl_reg,CTRL_REG_LINERELAY_BIT) &&
+		 BIT( port1,PORT_1_MDM_RTS_BIT) &&
+		!BIT( last_ctrl_reg,CTRL_REG_DTMF_BIT) &&
+		!BIT( last_ctrl_reg,CTRL_REG_MCBC_BIT) )
+	{
+		tonedetect = 1;
+	}
+	else
+	{
+		tonedetect = 0;
+	}
+
+	// Main transmission mode :  PORT_1_MDM_RTS = 0, CTRL_REG_DTMF = 1, CTRL_REG_MCBC = 0
+	if( BIT(last_ctrl_reg, CTRL_REG_LINERELAY_BIT) )
```

Comment:
Please don’t be inconsistent with spaces after commas like this.  We generally use space after comma.

Reply:
fixed